src/networkNode.ts

```diff
@@ -254,6 +254,79 @@ export class NetworkNode implements NetworkNodeInterface {
     }
   }
 
+
+  async getHistogramSamplesInBuckets(
+    rawmetricName: string,
+    buckets: string[], // empty string means all.
+    desiredMetricValue: number | null = null,
+    timeout = DEFAULT_INDIVIDUAL_TEST_TIMEOUT
+  ): Promise<number> {
+    let limitTimeout;
+    let expired: boolean = false;
+    try {
+      limitTimeout = setTimeout(() => {
+        debug(`Timeout getting metric ${rawmetricName} (${timeout})`);
+        expired = true;
+      }, timeout * 1000);
+
+      const metricName = getMetricName(rawmetricName);
+      let histogramBuckets = await getHistogramBuckets(this.prometheusUri, metricName);
+      let value = this._getSamplesCount(histogramBuckets, buckets);
+      if (desiredMetricValue === null || value >= desiredMetricValue) {
+        debug(`value: ${value} ~ desiredMetricValue: ${desiredMetricValue}`);
+        clearTimeout(limitTimeout);
+        return value;
+      }
+
+      // loop until get the desired value or timeout
+      let done = false;
+      let c = 0;
+      while (!done) {
```

Comment:
Would this scale well when we have grouping and 100+ nodes ?

Reply:
We will need to change some of the logic for `group` testing, is part of the refactoring that I'm working.
Thx!